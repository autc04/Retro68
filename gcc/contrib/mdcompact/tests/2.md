(define_insn "*movti_aarch64"
  [(set (match_operand:TI 0
	 "nonimmediate_operand"  "=   r,w,w,w, r,w,r,m,m,w,m")
	(match_operand:TI 1
	 "aarch64_movti_operand" " rUti,Z,Z,r, w,w,m,r,Z,m,w"))]
  "(register_operand (operands[0], TImode)
    || aarch64_reg_or_zero (operands[1], TImode))"
  "@
   #
   movi\\t%0.2d, #0
   fmov\t%d0, xzr
   #
   #
   mov\\t%0.16b, %1.16b
   ldp\\t%0, %H0, %1
   stp\\t%1, %H1, %0
   stp\\txzr, xzr, %0
   ldr\\t%q0, %1
   str\\t%q1, %0"
  [(set_attr "type" "multiple,neon_move,f_mcr,f_mcr,f_mrc,neon_logic_q, \
		             load_16,store_16,store_16,\
                             load_16,store_16")
   (set_attr "length" "8,4,4,8,8,4,4,4,4,4,4")
   (set_attr "arch" "*,simd,*,*,*,simd,*,*,*,fp,fp")]
)
