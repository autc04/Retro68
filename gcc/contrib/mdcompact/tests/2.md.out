(define_insn "*movti_aarch64"
  [(set (match_operand:TI 0
	 "nonimmediate_operand")
	(match_operand:TI 1
	 "aarch64_movti_operand"))]
  "(register_operand (operands[0], TImode)
    || aarch64_reg_or_zero (operands[1], TImode))"
  {@ [ cons: =0 , 1    ; attrs: type  , length , arch  ]
     [ r        , rUti ; multiple     , 8      , *     ] #
     [ w        , Z    ; neon_move    , 4      , simd  ] movi\t%0.2d, #0
     [ w        , Z    ; f_mcr        , 4      , *     ] fmov\t%d0, xzr
     [ w        , r    ; f_mcr        , 8      , *     ] #
     [ r        , w    ; f_mrc        , 8      , *     ] #
     [ w        , w    ; neon_logic_q , 4      , simd  ] mov\t%0.16b, %1.16b
     [ r        , m    ; load_16      , 4      , *     ] ldp\t%0, %H0, %1
     [ m        , r    ; store_16     , 4      , *     ] stp\t%1, %H1, %0
     [ m        , Z    ; store_16     , 4      , *     ] stp\txzr, xzr, %0
     [ w        , m    ; load_16      , 4      , fp    ] ldr\t%q0, %1
     [ m        , w    ; store_16     , 4      , fp    ] str\t%q1, %0
  }
)
